test/read-file.test.js

```diff
@@ -101,77 +155,155 @@ describe('+ readFile()', function () {
         bothDone = true
       })
     })
+
+    it('should reject the promise', function (done) {
+      jf.readFile(file)
+        .then(data => { })
+        .catch(err => {
+          assert(err instanceof Error)
+          assert(err.message.match(fn))
+          done()
+        })
+    })
   })
 
   describe('> when JSON reviver is set', function () {
-    it('should transform the JSON', function (done) {
-      var file = path.join(TEST_DIR, 'somefile.json')
-      var sillyReviver = function (k, v) {
+    var file, sillyReviver
+
+    beforeEach(function (done) {
+      file = path.join(TEST_DIR, 'somefile.json')
+      sillyReviver = function (k, v) {
         if (typeof v !== 'string') return v
         if (v.indexOf('date:') < 0) return v
         return new Date(v.split('date:')[1])
       }
 
-      var obj = {
-        name: 'jp',
-        day: 'date:2015-06-19T11:41:26.815Z'
-      }
+      var obj = { name: 'jp', day: 'date:2015-06-19T11:41:26.815Z' }
 
       fs.writeFileSync(file, JSON.stringify(obj))
-      jf.readFile(file, {reviver: sillyReviver}, function (err, data) {
+      done()
+    })
+
+    it('should transform the JSON', function (done) {
+      jf.readFile(file, { reviver: sillyReviver }, function (err, data) {
         assert.ifError(err)
         assert.strictEqual(data.name, 'jp')
         assert(data.day instanceof Date)
         assert.strictEqual(data.day.toISOString(), '2015-06-19T11:41:26.815Z')
         done()
       })
     })
+
+    it('should resolve the promise with transformed JSON', function (done) {
+      jf.readFile(file, { reviver: sillyReviver })
+        .then(data => {
+          assert.strictEqual(data.name, 'jp')
+          assert(data.day instanceof Date)
+          assert.strictEqual(data.day.toISOString(), '2015-06-19T11:41:26.815Z')
+          done()
+        }).catch(err => {
+          assert.ifError(err)
+          done()
+        })
+    })
   })
 
   describe('> when passing null and callback', function () {
```

Comment:
Need to edit this description to `> when passing null as options`.

Reply:
done